src/datastore/postgres-store.ts

```diff
@@ -1878,7 +1878,7 @@ export class PgDataStore
       }
       const txQuery = await client.query<{ tx_id: Buffer }>(
         `
-        SELECT tx_id
+        SELECT DISTINCT(tx_id)
```

Comment:
Just a `distinct` here seems like a bandaid, shouldn't we try to prevent the query from returning the duplicate transactions with a WHERE clause? Most likely needs `where canonical = true and microblock_canonical = true`.